library/alloc/tests/boxed.rs

```diff
@@ -3,6 +3,25 @@ use core::cell::Cell;
 use core::mem::MaybeUninit;
 use core::ptr::NonNull;
 
+#[test]
+fn box_partial_eq_different_allocators() {
+    struct AnotherAllocator(ConstAllocator);
+
+    unsafe impl Allocator for AnotherAllocator {
+        fn allocate(&self, layout: Layout) -> Result<NonNull<[u8]>, AllocError> {
+            self.0.allocate(layout)
+        }
+
+        unsafe fn deallocate(&self, ptr: NonNull<u8>, layout: Layout) {
+            unsafe { self.0.deallocate(ptr, layout) }
+        }
+    }
+
+    let b1 = Box::new_in(42, ConstAllocator);
```

Comment:
Is it ok to use `ConstAllocator` from this module for this test, or should this test be made in a different way? Or removed altogether?

Reply:
Ah, judging by a failing test, I guess it isn't.

Reply:
I've removed this test for the time being to check that others pass. I guess it should be possible to force type checking without having actual allocations within the test. Would it be useful to have such a test?